src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VariantsMongoReader.java

```diff
@@ -99,16 +108,36 @@ protected void doOpen() throws Exception {
     }
 
     @Override
-    protected EnsemblVariant doRead() throws Exception {
-        DBObject dbObject = delegateReader.doRead();
-        if (dbObject != null) {
-            SimplifiedVariant variant = converter.read(SimplifiedVariant.class, dbObject);
-            return buildVariantWrapper(variant);
+    protected List<EnsemblVariant> doRead() throws Exception {
+        if (moreElementsAvailable) {
```

Comment:
I think that this function should be simplified and avoid the flag variable. Also the read could probably be done with a batch size and avoid the roundtrips for each element and still use the spring converter.